### PR TITLE
#379 add CartDiscount benchmark tests

### PR DESCRIPTION
**Pull request by ahmetoz**
#### Summary
add CartDiscount benchmark tests

### CODE REVIEW

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+class CartDiscountSyncBenchmark {
+
+    private CartDiscountSyncOptions cartDiscountSyncOptions;
+    private List<String> errorCallBackMessages;
+    private List<String> warningCallBackMessages;
+    private List<Throwable> errorCallBackExceptions;
+
+    @BeforeAll
+    static void setup() {
+        deleteTypes(CTP_TARGET_CLIENT);
```
Comment (heshamMassoud) [inline src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: why do we need to delete types?

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+
+        // benchmark
+        final long beforeSyncTime = System.currentTimeMillis();
+        // Update cart discount drafts to target project with different stacking mode (STACKING)
+        final CartDiscountSyncStatistics syncStatistics = executeBlocking(cartDiscountSync.sync(cartDiscountDrafts));
+        final long totalTime = System.currentTimeMillis() - beforeSyncTime;
+
+        // assert on threshold (based on history of benchmarks; highest was ~13 seconds)
+        final int threshold = 26000; // double of the highest benchmark
```
Comment (heshamMassoud) [inline src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: I guess this is not true for this benchmark yet ;)

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+        // benchmark
+        final long beforeSyncTime = System.currentTimeMillis();
+        // Create half of the cart discount drafts to target project
+        // and update half of the cart discount drafts to target project with different stacking mode (STACKING)
+        final CartDiscountSyncStatistics syncStatistics = executeBlocking(cartDiscountSync.sync(cartDiscountDrafts));
+        final long totalTime = System.currentTimeMillis() - beforeSyncTime;
+
+        // assert on threshold (based on history of benchmarks; highest was ~12 seconds)
+        final int threshold = 24000; // double of the highest benchmark
```
Comment (heshamMassoud) [inline src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: I guess this is not true for this benchmark yet ;)

File: src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java [hunk 4]
```diff
@@ -138,7 +138,7 @@ public static void populateTargetProject() {
         int counter = 0;
         int current = 1;
 
-        while (counter != 100) {
+        while (counter != capacity) {
```
Comment (heshamMassoud) [inline src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java]: is this change related to this PR?

Comment (ahmetoz) [reply src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java]: here I am passing capacity as a parameter but not using it :) I realized this problem when I needed to create sort orders more than 100 :)

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java (hunk 2 above)
Comment (ahmetoz) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: what is your suggestion and how did you calculate these things before benchmarks?

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java (hunk 1 above)
Comment (ahmetoz) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: CartDiscounts could refer to the custom fields (Types).

Comment (heshamMassoud) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: but it doesn't seem that you are using custom fields for the cart discounts you are creating in the benchmarks.

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java (hunk 2 above)
Comment (heshamMassoud) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: you can run it locally on your machine. Get the time it took to finish for each benchmark and use its double as the threshold.

File: src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java (hunk 4 above)
Comment (ahmetoz) [reply src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java]: 698c1f5

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java (hunk 1 above)
Comment (ahmetoz) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: 2383e8f

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java (hunk 2 above)
Comment (ahmetoz) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: 50527df

File: src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java (hunk 3 above)
Comment (ahmetoz) [reply src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java]: 50527df
